modules/custom/az_migration/src/Plugin/migrate/process/ViewsReferenceMapping.php

```diff
@@ -0,0 +1,208 @@
+   *
+   * @var array
+   */
+  const VIEW_MAPPING = [
+    'uaqs_events' => [
+      'view' => 'az_events',
+      'display' => [
+        'default' => 'page_1',
+        'page' => 'page_1',
+        'list_block' => 'page_1',
+        'card_group_block' => 'az_grid',
+        'block_1' => 'az_sidebar',
+      ],
+      'argument_migrations' => [
+        'az_event_categories',
+      ],
+    ],
+    'uaqs_news' => [
+      'view' => 'az_news',
+      'display' => [
+        'default' => 'az_grid',
+        'three_col_news_block_3' => 'az_grid',
+        'three_col_news_block' => 'az_grid',
+        'sidebar_promoted_news' => 'az_sidebar',
+        'uaqs_teaser_list_page' => 'az_teaser_grid',
+        'uaqs_media_list_page' => 'az_paged_row',
+        'recent_news_marquee' => 'marquee',
+        'recent_news_medium_media_list' => 'az_paged_row',
+      ],
+      'argument_migrations' => [
+        'az_news_tags',
+      ],
+    ],
+    'uaqs_person_directory' => [
+      'view' => 'az_person',
+      'display' => [
+        'default' => 'grid',
+        'page' => 'grid',
+        'page_1' => 'row',
+      ],
+      'argument_migrations' => [
+        'az_person_categories',
+        'az_person_categories_secondary',
+      ],
+    ],
+    'uaqs_content_chunks_views_page_by_category' => [
+      'view' => 'az_page_by_category',
+      'display' => [
+        'default' => 'row',
+        'page' => 'row',
+        'page_1' => 'grid',
+      ],
+      'argument_migrations' => [
+        'az_flexible_page_categories',
+      ],
+    ],
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+    );
+
+    $instance->migrateLookup = $container->get('migrate.lookup');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    $transformedValue = [];
+    $viewData = [];
+    $viewMapping = self::VIEW_MAPPING;
+
+    if (!empty($this->configuration['view_mapping']) && is_array($this->configuration['view_mapping'])) {
+      $viewMapping = array_merge($viewMapping, $this->configuration['view_mapping']);
+    }
+
+    // Get view and display name.
+    $viewDisplay = explode("|", $value['vname']);
+    $view = $viewDisplay[0];
+    $display = $viewDisplay[1];
+
+    if (isset($viewMapping[$view])) {
+      $transformedValue['target_id'] = $viewMapping[$view]['view'];
+      $transformedValue['display_id'] = $viewMapping[$view]['display'][$display];
+      $argumentMigrations = $viewMapping[$view]['argument_migrations'];
+    }
+    else {
+      $transformedValue['target_id'] = $view;
+      $transformedValue['display_id'] = $display;
+      $argumentMigrations = NULL;
+    }
```

Comment:
Question for team:  If a mapping does not exist for a source view reference, should we skip the row/entity containing the viewsfield, e.g. the Views Paragraph?  If not, the source values are copied directly so we end up with a broken views reference.

Another possible option: create a migrate stub intead?

Reply:
I think we can skip the row and add a migrate message like the skip_on_empty plugin allows for.
https://api.drupal.org/api/drupal/core%21modules%21migrate%21src%21Plugin%21migrate%21process%21SkipOnEmpty.php/class/SkipOnEmpty/9.3.x

Reply:
This would allow rerunning the migration with the --update flag after creating the view on the destination site, and mapping it in their migration.

Reply:
I do love validation. You might have to create your own custom constraint for this since exiting ones seem to require permissions https://www.berramou.com/blog/drupal-8-create-custom-validation-constraint

https://gitlab.si.umich.edu/root/test/-/blob/8.5.x/core/lib/Drupal/Core/Entity/Plugin/Validation/Constraint/ValidReferenceConstraint.php

Reply:
I was just thinking we could try loading the view with the EntityTypeManager service for validation.

Reply:
The constraint I linked uses the entityTypeManager

Reply:
FYI an alternative approach would be to add the validation flag to node migrations, which would run field validation for everything, but that would require that we migrate users with a higher level permission unfortunately, since node validation takes permissions into account.

Reply:
Looks like since the destination field type extends core's `EntityReferenceItem` field type which already uses the `ValidReferenceConstraint` so I think adding the validation flag to the migration file would effectively force that constraint to be used.

Reply:
In the case where a mapping doesn't exist, I think it would be sufficient to skip the row with a migrate message notifying about the failed mapping value.

Reply:
Thanks all.